docs/custom-component.md

```diff
@@ -53,4 +53,27 @@ You can read more about the inject/provide feature on the [Vue.js documentation]
 
 The search store will be automatically fetched from a parent [Index component](/components/index.md) higher in the DOM tree.
 
-As a reminder, you can also [provide your custom searchStore to any custom component](using-components.md#manually-inject-the-search-store-into-components) as a property.
+As a reminder, you can also [provide your custom searchStore to any custom component](components.md#manually-inject-the-search-store-into-components) as a property.
+
+## Best practices
+
+To ensure consistency and re-usability for custom components, we recommend reviewing the following guidelines.
+
+### Styles and CSS classes
+
+* Vue InstantSearch uses `ais` as a prefix for CSS classes, we recommend choosing a different prefix to avoid conflicts
+* We do not recommend the use of [`scoped`](https://vue-loader.vuejs.org/en/features/scoped-css.html) styles, it makes it very hard to override them.
+* Use the [BEM notation](http://getbem.com/introduction/) with only one depth level.
+* Think about reusability, ship the bare minimum style for your component to be disaplayed well. While allowing for easy customization.
+
+### Vue component
+
+* Use the `Component` mixin that we provide. This will make sure your component can resolve the `searchStore` if not provided. It ensures the `searchStore` prop is available in your component at any time.
+* If you need to mutate the `searchStore` multiple times, please use `searchStore.stop()` and `searchStore.start()`, so that other components don't update their rendering on every intermediary state mutation.
+* Make sure that when the component is mounted, you catch up with the `searchStore`. You can optionally mutate the state of the `searchStore` at this stage.
+* When a component is `unmounted` or `destroyed`, make sure that you leave the `searchStore` in a state that does not include things you might have added (facets / filters / etc.).
+* Make sure your component gracefully handles any state of the `searchStore`.
+
+### Export UMD + ES2015
+
+* Export your components as UMD, CommonJS and ES2015. @TODO: this needs a link or a blog post from Raymond, otherwise people will not know how to do that properly.
```

Comment:
As soon as we give this documentation to end users, if we have parts that are confusing or that people will not be able to act on (like this one) then we either need to remove it or just link to an explanation/way to do it.

Otherwise this can be frustrating. YOU know how to do "export UMD + ES2015". I don't know at all.